### DISCUSSION TITLE
how to change ike-port and nat-T port , android-->nat-->ipsec gw？

### DISCUSSION THREAD

**Discussion by ygm521** [strongswan/strongswan#998] [General]
how to change ike-port and nat-T port , android-->nat-->ipsec gw？ how config？ thanks！

**Comment by tobiasbrunner** [comment]
Please see [here](https://docs.strongswan.org/strongswan-docs/5.9/features/natTraversal.html#_custom_server_ports).

**Comment by ygm521** [comment]
ipsec gw:
  charon.port_nat_t=14500
  charon.port=1500

android:
  charon.port_nat_t=14500
  charon.port=1500
  charon.connectin.remote_port=1500

ike succuss,but not ping protect NET;

**Reply by tobiasbrunner** [reply]
On the client, I'd recommend setting `port_nat_t` and `port` to `0` in order to use ephemeral source ports (that's already the case in our Android app). And I'm not sure what exactly `charon.connectin.remote_port` refers to, even with the typo fixed I'm not aware of any such option.

Regarding the other issue, please refer to #196.
